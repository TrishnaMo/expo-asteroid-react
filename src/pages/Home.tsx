import React, { useState } from "react";
import { Container, Box, TextField, Button } from "@material-ui/core";
import { useHistory } from "react-router-dom";

const Home = () => {
  const [asteroidId, setAsteroidId] = useState<string>("");
  const history = useHistory();

  const navigateToAsteroidDetails = (id: string) => {
    history.push(`/asteriod-details/${id}`);
  };

  return (
    <Container
      component="main"
      style={{
        display: "flex",
        height: "100%",
        alignItems: "center",
      }}
    >
      <Container maxWidth="xs">
        <Box
          style={{
            flex: 1,
            flexDirection: "column",
            alignItems: "center",
          }}
        >
          <Box
            style={{
              border: "1px solid #ccc",
              padding: "35px 30px",
              borderRadius: 6,
            }}
          >
            <TextField
              fullWidth
              id="asteroidId"
              data-testid="asteroidId"
              label="Enter Asteroid ID"
              name="asteroidId"
              autoFocus
              variant="outlined"
              onChange={(e) => {
                setAsteroidId(e.target.value);
              }}
            />
            <Button
              type="submit"
              data-testid="submit"
              fullWidth
              variant="contained"
              color="primary"
              style={{ marginTop: 35, padding: 12 }}
              disabled={asteroidId === ""}
              onClick={() => {
                navigateToAsteroidDetails(asteroidId);
              }}
            >
              Submit
            </Button>
          </Box>
          <Button
            type="button"
            data-testid="getRandomAsteroid"
            fullWidth
            variant="contained"
            color="primary"
            style={{ marginTop: 35, padding: 12 }}
            onClick={() => {
              navigateToAsteroidDetails('random');
            }}
          >
            Random Asteroid
          </Button>
        </Box>
      </Container>
    </Container>
  );
};

export default Home;
