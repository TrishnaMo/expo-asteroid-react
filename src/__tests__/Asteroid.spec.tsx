import React from "react";
import { render } from "@testing-library/react";
import Home from "../pages/Home";

describe("Test Home Page", () => {
  it("renders snapshot as expected", () => {
    const home = render(<Home />);
    expect(home).toMatchSnapshot();
  });

  it("should show submit button", () => {
    const { getByTestId } = render(<Home />);
    expect(getByTestId("submit").textContent).toBe("Submit");
  });
});
