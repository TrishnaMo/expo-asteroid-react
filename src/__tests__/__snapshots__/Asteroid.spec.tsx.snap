// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home Page renders snapshot as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="MuiContainer-root MuiContainer-maxWidthLg"
        style="display: flex; height: 100%; align-items: center;"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthXs"
        >
          <div
            class="MuiBox-root MuiBox-root-1"
            style="flex: 1; flex-direction: column; align-items: center;"
          >
            <div
              class="MuiBox-root MuiBox-root-2"
              style="border: 1px solid #ccc; padding: 35px 30px; border-radius: 6px;"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                data-testid="asteroidId"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused"
                  data-shrink="true"
                  for="asteroidId"
                  id="asteroidId-label"
                >
                  Enter Asteroid ID
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    id="asteroidId"
                    name="asteroidId"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-5 PrivateNotchedOutline-legendNotched-6"
                    >
                      <span>
                        Enter Asteroid ID
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled MuiButton-fullWidth Mui-disabled"
                data-testid="submit"
                disabled=""
                style="margin-top: 35px; padding: 12px;"
                tabindex="-1"
                type="submit"
              >
                <span
                  class="MuiButton-label"
                >
                  Submit
                </span>
              </button>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
              data-testid="getRandomAsteroid"
              style="margin-top: 35px; padding: 12px;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Random Asteroid
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="MuiContainer-root MuiContainer-maxWidthLg"
      style="display: flex; height: 100%; align-items: center;"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXs"
      >
        <div
          class="MuiBox-root MuiBox-root-1"
          style="flex: 1; flex-direction: column; align-items: center;"
        >
          <div
            class="MuiBox-root MuiBox-root-2"
            style="border: 1px solid #ccc; padding: 35px 30px; border-radius: 6px;"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              data-testid="asteroidId"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused"
                data-shrink="true"
                for="asteroidId"
                id="asteroidId-label"
              >
                Enter Asteroid ID
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="asteroidId"
                  name="asteroidId"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-5 PrivateNotchedOutline-legendNotched-6"
                  >
                    <span>
                      Enter Asteroid ID
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled MuiButton-fullWidth Mui-disabled"
              data-testid="submit"
              disabled=""
              style="margin-top: 35px; padding: 12px;"
              tabindex="-1"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Submit
              </span>
            </button>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            data-testid="getRandomAsteroid"
            style="margin-top: 35px; padding: 12px;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Random Asteroid
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
